pkg/pillar/pubsub/pubsub.go

```diff
@@ -163,29 +133,48 @@ func (p *PubSub) subscribeImpl(agentName string, agentScope string, topicImpl in
 // publishImpl init function to create directory and socket listener based on above settings
 // We read any checkpointed state from dirName and insert in pub.km as initial
 // values.
-func (p *PubSub) publishImpl(agentName string, agentScope string,
-	topicType interface{}, persistent bool) (Publication, error) {
 
-	topic := TypeToName(topicType)
+// PublicationOptions defines all the possible options a new publication may have
+type PublicationOptions struct {
+	AgentName  string
+	AgentScope string
+	TopicType  interface{}
+	Persistent bool
+}
+
+// NewPublication creates a new publication with given options
```

Comment:
Small nit, but should capitalize: `creates a new Publication` (that way it is clear that we are referring to the interface returned)